### PR TITLE
sudoswap: price fix and fee move into new column, break out token_id values

### PR DESCRIPTION
**Pull request by andrewhong5297**
Brief comments on the purpose of your changes:

For Dune Engine V2
I've checked that:

 I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
 I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables
 if adding a new model, I added a test
 the filename is unique and ends with .sql
 each sql file is a select statement and has only one view, table or function defined
 column names are lowercase_snake_cased
When you are ready for a review, tag duneanalytics/data-experience. We will re-open your forked pull request as an internal pull request. Then your spells will run in dbt and the logs will be avaiable in Github Actions DBT Slim CI. This job will only run the models and tests changed by your PR compared to the production project.

---

Per @0xRobin comments on https://github.com/duneanalytics/abstractions/pull/1364, I'm moving owner fees to their own column and nulling the royalty columns.

Also including fees into trade prices, mainly on sell trades since buy trades already had fees included. Rob, would love any comments/suggestions you have before we merge again :)

### CODE REVIEW

File: spellbook/models/sudoswap/ethereum/sudoswap_ethereum_events.sql [hunk 1]
```diff
@@ -211,9 +215,16 @@ WITH
             --royalties don't technically exist on AMM, but there are owner fees for the pool that can be treated as royalties in the future.
             , base_price*ownerfee as royalty_fee_amount_raw
```
[the remark this answers is not in the document]
Comment (andrewhong5297) [reply spellbook/models/sudoswap/ethereum/sudoswap_ethereum_events.sql]: ah yeah good catch!

**Review by jeff-dude — APPROVED** [review]

**Comment by 0xRobin** [comment]
@andrewhong5297 I still think the math might be off.
Let me pull this locally and open up a new PR.

**Comment by andrewhong5297** [comment]
> @andrewhong5297 I still think the math might be off.
> Let me pull this locally and open up a new PR. (once the repo changes have been put in place today)

Sg, thanks for taking a look!

**Comment by andrewhong5297** [comment]
Buy price is def fine, sell price remember that owner fees is subtracted too do that has to be added back - your queries don't account for that I believe (may be wrong).

**Comment by andrewhong5297** [comment]
ah I didn't do `+ owner_fee*base_price` on the sell prices I think. lmk if you find any other errors!

**Comment by andrewhong5297** [comment]
doing some more QA, I think there are some txs where only the first swap event prices were captured. currently debugging, will circle back (https://dune.com/queries/1199101?d=1)

**Comment by 0xRobin** [comment]
yeah, also refunds will break the current calculation. Not sure if they already happened on non-zero swaps though. (partial fills)

**Comment by 0xRobin** [comment]
I'm working the math out again from the start at the moment. We should be able to make something soundproof if we combine later. 🙏

**Comment by andrewhong5297** [comment]
> yeah, also refunds will break the current calculation. Not sure if they already happened on non-zero swaps though. (partial fills)

Ah had no idea about refunds. Do you have any links or examples?

**Comment by andrewhong5297** [comment]
> I'm working the math out again from the start at the moment. We should be able to make something soundproof if we combine later. 🙏

awesome, dm me on twitter in case I don't respond here. Lmk if it'd be easier to hop on a call or something and just jam on it (or we could livestream debug it haha).

**Comment by 0xRobin** [comment]
> Ah had no idea about refunds. Do you have any links or examples?

There's 2 types of refunds happening: router lvl and pool lvl. 
https://github.com/sudoswap/lssvm/blob/f970d9d2a2fc03a4409a2d399792d1d8bf2c2d66/src/LSSVMRouter.sol#L939-L941
https://github.com/sudoswap/lssvm/blob/f970d9d2a2fc03a4409a2d399792d1d8bf2c2d66/src/LSSVMPair.sol#L173

With the way we filter the traces currently, only pool lvl refunds should really be a concern?
I'm trying to find if any already happened on chain though

**Comment by andrewhong5297** [comment]
> > Ah had no idea about refunds. Do you have any links or examples?
> 
> There's 2 types of refunds happening: router lvl and pool lvl. https://github.com/sudoswap/lssvm/blob/f970d9d2a2fc03a4409a2d399792d1d8bf2c2d66/src/LSSVMRouter.sol#L939-L941 https://github.com/sudoswap/lssvm/blob/f970d9d2a2fc03a4409a2d399792d1d8bf2c2d66/src/LSSVMPair.sol#L173
> 
> With the way we filter the traces currently, only pool lvl refunds should really be a concern? I'm trying to find if any already happened on chain though

ah I see. maybe just adding a filter for `tr.to != tx_from` will be good enough to filter out refunds? That does mean we'll have to join `ethereum.transactions` earlier rather than at the end.

**Comment by 0xRobin** [comment]
oh I think we'll need to make sure we do include them in the calculations but **substract** them, since the amount originally payed to a pool will be trade_price + refund.

I believe we can check on tr.to = call_from so no need to pull in the transactions yet.

**Comment by andrewhong5297** [comment]
ahh I'm following now. hmm okay...

**Comment by andrewhong5297** [comment]
okay it seems some fees/trades are missing (~1%) because of the `tr.to != asset_recip` filter in the events table. I remember there was some case where keeping `asset_recip` was causing a double count because it would transfer ETH to the contract and then to the recip, but I can't find any examples anymore. will circle back.

**Comment by 0xRobin** [comment]
oh good catch!

This was something that I also stumbled on but didn't think the edge case would really be important at this stage :sweat_smile: 
So in the example transaction you linked:
The user sold 4 NFTs
3 to pool 0xb14
1 to pool 0x000 (where the caller itself is the recipient, hance the NFT transfer to itself).
The traces here in that example should not really be an edge case though?

Would like to see the reverse BUY case, where that indeed makes it difficult. 
Also, in that case it will be almost impossible to distinguish between ETH that gets payed to the recipient of the pool and ETH that gets refunded. :cold_sweat:

**Comment by 0xRobin** [comment]
Gonna push the changes tomorrow, think I have a decent workaround if filter the traces differently with the trade type in mind. I'll tag you in the PR

**Comment by jeff-dude** [comment]
please keep in mind the repo structure change prior to new PRs.
1. changed repo name from 'abstractions' to 'spellbook'
2. renamed default branch from 'master' to 'main'
3. moved all v1 engine abstractions into 'deprecated-dune-v1-abstractions'
4. removed 'spellbook' directory at root and moved all objects within (i.e. models directory you use here) to the root level

it might be easiest to start fresh with a fork of the repo and reset local env.

**Comment by 0xRobin** [comment]
> please keep in mind the repo structure change prior to new PRs.

Yes, I noticed the PR was merged so all changes are in effect now, right?
will pull a new branch from main :+1:

**Pull request merged**